### PR TITLE
Create Sprint 6 Preparation Plan for EPIC 2

### PR DESCRIPTION
**Pull request by jeffreyhorn**
Add comprehensive PREP_PLAN.md with 10 prioritized tasks:

1. Create Sprint 6 Known Unknowns List (Critical)
   - 18-25 unknowns across 4 categories
   - Covers convexity, bug fixes, GAMSLib, UX

2. Research Convexity Detection Approaches (Critical)
   - POC pattern matchers for heuristic detection
   - Test suite with convex/non-convex models
   - Recommendation: Approach 1 vs Approach 2

3. Analyze Maximize Bug Root Cause (Critical)
   - Minimal test cases isolating the bug
   - KKT theory validation for maximize
   - Fix design with 3 options evaluated

4. Survey GAMSLib NLP Model Catalog (High)
   - 30-50 models cataloged
   - Parser feature matrix
   - Tier 1 selection (10 models for Sprint 6)

5. Design Nested Min/Max Flattening Strategy (High)
   - Option A flattening algorithm
   - 4+ test cases
   - 4-6 hour implementation estimate

6. Prototype Error Message Improvements (High)
   - Enhanced error template design
   - Formatter implementation
   - Integration plan

7. Set Up GAMSLib Download Infrastructure (Medium)
   - Automated download script
   - Model manifest and README
   - Idempotent execution

8. Create Convexity Test Fixtures (Medium)
   - 10-15 test models (convex, non-convex, edge cases)
   - Expected results documented

9. Audit Current Test Coverage (High)
   - Baseline: 1078 tests, 87% coverage
   - Gap analysis for <90% modules
   - CI/CD coverage enforcement

10. Plan Sprint 6 Detailed Schedule (Critical)
    - Day-by-day breakdown (10 days)
    - 3 checkpoints with criteria
    - Risk register

Total estimated time: 34-48 hours (4-6 working days) All tasks follow SPRINT_4 and SPRINT_5 PREP_PLAN format.

### CODE REVIEW

File: docs/planning/EPIC_2/SPRINT_6/PREP_PLAN.md [hunk 1]
```diff
@@ -0,0 +1,1502 @@
...
+                    collected.append(arg)
+            
+            return Call("min", collected)
+        
+        case Call("max", args):
+            # Similar logic for max
+            ...
+        
+        case _:
+            return expr
```
[the remark this answers is not in the document]
Comment (jeffreyhorn) [reply docs/planning/EPIC_2/SPRINT_6/PREP_PLAN.md]: Replace match/case with if/elif for Python compatibility
   - Remove Python 3.10+ structural pattern matching
   - Use compatible if/elif syntax for broader support
   - Add note about Python 3.11+ project requirement

File: docs/planning/EPIC_2/SPRINT_6/PREP_PLAN.md [hunk 2]
```diff
@@ -0,0 +1,1502 @@
...
+#### Step 4: Design Fix (1-2 hours)
+
+Based on analysis, choose fix strategy:
+
+**Option A: Flip bound multiplier signs**
+```python
+if objective_sense == ObjSense.MAXIMIZE:
+    # Flip signs for maximize
+    expr = Binary("+", expr, MultiplierRef(piL_name, ...))  # was -
+    expr = Binary("-", expr, MultiplierRef(piU_name, ...))  # was +
```
[the remark this answers is not in the document]
Comment (jeffreyhorn) [reply docs/planning/EPIC_2/SPRINT_6/PREP_PLAN.md]: Use descriptive placeholders in MultiplierRef examples
   - Replace '...' with 'var_indices' 
   - Add comment explaining parameter purpose

File: docs/planning/EPIC_2/SPRINT_6/PREP_PLAN.md [hunk 3]
```diff
@@ -0,0 +1,1502 @@
...
+# Run POC on test suite
+python scripts/poc_convexity_patterns.py tests/fixtures/convexity/*.gms
+
+# Expected results:
+# - convex_lp.gms: 0 warnings
+# - convex_qp.gms: 0 warnings
+# - nonconvex_circle.gms: 1 warning (nonlinear equality)
+# - nonconvex_trig.gms: 2 warnings (trig + nonlinear eq)
+# - nonconvex_bilinear.gms: 1 warning (bilinear)
+# - convex_with_nonlinear_ineq.gms: 0 warnings (inequalities OK)
```
[the remark this answers is not in the document]
Comment (jeffreyhorn) [reply docs/planning/EPIC_2/SPRINT_6/PREP_PLAN.md]: Use structured YAML manifest for expected test results
   - Replace inline comments with tests/fixtures/convexity/expected_results.yaml
   - Enables automated validation of POC results
